rationale/message_block_layout.md

```diff
@@ -46,4 +46,4 @@ The last piece of the puzzle is determining _which_ row the message is placed at
 
 ### Caveats
 
-The message placement rules described above conflict with the first rule that shares must be ordered by namespace ID, as shares between two messages that are not placed adjacent to each other do not have a natural namespace they belong to. This is resolved by having special transactions the block producer includes that specify a range a zero-padding shares for a given namespace ID (which must be between the namespace IDs of the surrounding real shares, inclusive).
+The message placement rules described above conflict with the first rule that shares must be ordered by namespace ID, as shares between two messages that are not placed adjacent to each other do not have a natural namespace they belong to. This is resolved by having [special transactions](./../specs/data_structures.md#signedtransactiondatapayforpadding) the block producer includes that specify a range a zero-padding shares for a given namespace ID (which must be between the namespace IDs of the surrounding real shares, inclusive).
```

Comment:
Does this mean that it becomes required to send this additional Tx? So if I want to include a message, I'll have to send a Tx that pays for a message and for padding?

Reply:
If that's the case, I think this introduces some undesirable complexity.

Reply:
also there seems to be a typo in 
> the block producer includes that specify a range a zero-padding shares 

How does the user know what range to pay for?

Reply:
These are created by the block proposer themselves, i.e. not broadcast through the p2p network. So there's no typo.

Reply:
I don't understand. The description for https://github.com/lazyledger/lazyledger-specs/pull/89/commits/6d9648a454bcbfef79172152a4f129253cb7962d#diff-6aed1ad746ae3318f74b4ee4094dda4664865e75bf4b8a3c95e58773292d1c74R603 says that this special transaction pays for padding. Who's paying for the padding if not the user, and how is it that block producers can generate these transactions themselves?

Reply:
The block proposer pays for padding?

>So there's no typo.
Shouldn't that be: that specify a range *of* zero-padding shares?

Reply:
> Who's paying for the padding if not the user

The block proposer. They know what's in the block so they can make new txs. Note that I haven't actually decided if a fee should be paid for these, and if not then there's not much point in having them as txs vs just part of the block data that notate inter-message padding. That's for a future issue.

Reply:
I find it weird that the block proposer would pay for these shares. Why wouldn't they simply drop all the relevant messages instead.

Reply:
Yeah, I'm not sure about the economic rationale behind that.

Reply:
> Yeah, I'm not sure about the economic rationale behind that.

Agree that it's not clear either way, hence me saying that not having this information as transactions is potentially the way to go, but it'll require further investigation that's outside the scope of this PR.

Reply:
Filed #98 to track this issue.